packages/tpetra/core/src/Tpetra_Distributor.hpp

```diff
@@ -2306,6 +2549,104 @@ namespace Tpetra {
         *out_ << os.str ();
       }
 
+      if (sendType == Details::DISTRIBUTOR_PERSISTENT) {
+
+        if (persistentRequests_.size() == 0) {
+
+          typedef typename ExpView::non_const_value_type Packet;
+          // typedef typename ExpView::array_layout Layout;
+          // typedef typename ExpView::device_type Device;
+          // typedef typename ExpView::memory_traits Mem;
+
+          Teuchos::RCP<const Teuchos::MpiComm<int> > tmpic = Teuchos::rcp_dynamic_cast<const Teuchos::MpiComm<int> >(comm_);
+          MPI_Comm rawComm = * (tmpic->getRawMpiComm ());
+
+          if (verbose_) {
+            std::ostringstream os;
+            os << "Proc " << myRank << ": Creating persistent recv requests" << endl;
+            *out_ << os.str ();
+          }
+          // create the persistent recv requests
+          size_t curBufOffset = 0;
+          for (size_type i = 0; i < actualNumReceives; ++i) {
+            const size_t curBufLen = lengthsFrom_[i] * numPackets;
+            if (procsFrom_[i] != myRank) {
+              imports_view_type recvBuf =
+                subview_offset (imports, curBufOffset, curBufLen);
+
+              Packet t;
+              MPI_Datatype rawType = Teuchos::Details::MpiTypeTraits<Packet>::getType (t);
+              Packet* rawRecvBuf = recvBuf.data ();
+              const int count = as<int> (recvBuf.size ());
+              MPI_Request rawRequest = MPI_REQUEST_NULL;
+              MPI_Recv_init (rawRecvBuf, count, rawType, procsFrom_[i], tag, rawComm, &rawRequest);
+
+              Teuchos::RCP<Teuchos::CommRequest<int> > req = Teuchos::mpiCommRequest<int>(rawRequest, count);
+              // Teuchos::set_extra_data( globalRecvBuf, "buffer", inOutArg(req) );
```

Comment:
The encapsulation should be able to take a buffer.  Remember that you can always wrap a `Kokkos::View` in the custom deleter for a `Teuchos::ArrayRCP`.  We have helper functions for that.